plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -775,6 +803,21 @@ private static LongWriteFunction dateWriteFunctionUsingLocalDate(ClickHouseVersi
         };
     }
 
+    private static ColumnMapping timestampColumnMapping(TimestampType timestampType, ClickHouseVersion version)
+    {
+        if (timestampType.getPrecision() <= TimestampType.MAX_SHORT_PRECISION) {
+            return ColumnMapping.longMapping(
+                    timestampType,
+                    timestampReadFunction(timestampType),
+                    shortTimestampWriteFunction(version));
+        }
+        checkArgument(timestampType.getPrecision() <= CLICKHOUSE_MAX_SUPPORTED_DATETIME64_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
```

Comment:
In other connectors we seem to perform rounding - see for example SQLServer where max supported precision is 7.

Reply:
Already performed rounding - see https://github.com/trinodb/trino/pull/15040/files#diff-7ce7bdaa32b38505167fecaa1621f0e71d942c962a7f39c0cce4ea5e21ef0a1aR855

The current code is similar to the SQLServer implementation:
- SQLServer calls `StandardColumnMappings#fromLongTrinoTimestamp`
- ClickHouse(my code) copies the code from `StandardColumnMappings#fromLongTrinoTimestamp`, adding a range check

What's confusing you I guess is the naming of the parameters, which should be changed from `precision` to `roundToPrecision`.